### PR TITLE
updated specification with explicit definition of inclusion criteria,…

### PR DESCRIPTION
**Pull request by chrissimpkins**
… deviations handling, formatted mandatory and optional specifications

This PR includes the following:

1. reformatted specifications into mandatory and optional components based upon my interpretation of your original set of specifications
2. explicit definitions of the "all" and "contributors" by contribution level/effort and contribution category
3. a spec for how projects should handle deviations from the "all" & "contributors" definitions as we discussed in IR #9 .

Take or leave any changes here.  Some of this is my interpretation of your original set of specs and I may have interpreted your intent/emphasis on icons over text (among other things) incorrectly.  The changes are intentionally verbose and are intended to leave little room for question about how "all" and "contributor" are defined in the specification.  The additions also provide project maintainers with an approach to handle deviations from the spec by contributor category and level of effort so that this is transparent to potential contributors.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,11 +16,25 @@ If you find maintaining the list of contributors tedious, try using the [CLI too
...
+
+### Optional
+Projects should consider the following optional items as part of their commitment to the All Contributors specification:
+
+1. An optional link from the Contribution Category indicator to the category-specific contribution(s) can be included to provide additional information about the project contribution(s) as detailed in the key below
+
+## Key to the Contribution Categories
```
Comment (kentcdodds) [inline README.md]: If you rename this heading, you'll need to update the link `emoji key` below.

Personally, I'd prefer to keep it `Emoji Key`. Short and simple. But I'm flexible.

**Comment by kentcdodds** [comment]
This is fantastic. Thank you for the contribution. Please address the one question/suggestion I have :+1:

**Comment by chrissimpkins** [comment]
Thanks Kent.  I changed it back to `Emoji key` so the link should be OK now.

**Comment by kentcdodds** [comment]
Fantastic thanks!

**Comment by chrissimpkins** [comment]
:+1:  great project!

**Comment by MatheusRV** [comment]
amazing update!!

**Pull request merged**